### PR TITLE
Week 2 HW Checkout debugging

### PR DESCRIPTION
**Pull request by lelandjordon**
Completeness: 3
Comfort: 2

Still getting lost when trying to figure out what refers to what.  Had to check the solution, too, when I tried replacing forEach with '=' everywhere and realized I'd have to change the included code -- which was the wrong plan for me to take.

(update) I'm going to have to relearn what "switch-case" does and why it's broken.  (I mistakenly called it "check-switch" in my commit remark!)  I think the rest of this code works now.

### CODE REVIEW

File: scripts/shopping_cart.js [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
 
-  add_item: function(name, price){
-    this.items.push([ name: name, cost: cost, weight: weight, size: size ]);
+  addItem: function(name, cost, weight, size, kind){
+    console.log("activate addItem function");
+    this.items.push({ name: name, cost: cost, weight: weight, size: size, kind: kind });
   },
 
   calculateShipping: function(item){
+    console.log("activate calculateShipping function");
```
Comment (andrewsunglaekim) [inline scripts/shopping_cart.js:15]: i really like these log statements that let you know when these functions are being called, i would change the sentence up a bit, maybe something like "calculateShipping function called"

File: scripts/shopping_cart.js [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
+    }
+    else if (weight <= 50){
       shipping += 5;
     } else {
-      shipping += 10
+      shipping += 10;
     }
 
-    switch(item.size){
+    switch(items.size){
```
Comment (andrewsunglaekim) [inline scripts/shopping_cart.js:27]: why did you decide on items vs item? Where is this variable coming from? Check out line 14. This function is using the argument `item` which is a variable we now have access to. We don't have access to `items`.

File: scripts/shopping_cart.js [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
+    console.log("activate addItem function");
+    this.items.push({ name: name, cost: cost, weight: weight, size: size, kind: kind });
   },
 
   calculateShipping: function(item){
+    console.log("activate calculateShipping function");
     var shipping = 0;
-    var weight = item.weight || 5;
-    if(weight < 50){
+    var weight = items.weight || 5;
```
Comment (andrewsunglaekim) [inline scripts/shopping_cart.js:17]: read comment below about `items` vs `item`

File: scripts/shopping_cart.js [hunk 4]
```diff
@@ -27,29 +42,37 @@ app.shoppingCart = {
...
+    var sub = 0.00;
     this.items.forEach(function(item){
-      cost = cost + item.cost;
-    })
-    return cost;
+      sub = sub + item.cost;
+      console.log(sub);
+    });
+    return sub;
   },
```
Comment (andrewsunglaekim) [inline scripts/shopping_cart.js:60]: :+1:

File: scripts/shopping_cart.js [hunk 5]
```diff
@@ -27,29 +42,37 @@ app.shoppingCart = {
...
 
   totalCost: function() {
-    return subtotal + total_shipping();
+    console.log("activate totalCost function");
+    return this.subtotal() + this.totalShipping();
   },
```
Comment (andrewsunglaekim) [inline scripts/shopping_cart.js:65]: :+1:

**Comment by andrewsunglaekim** [comment]
:+1:  Good job on this HW leland. There are some comments I'd like you to review. Additionally, since you referred to using the solution. I encourage you to re clone this exercise into a different directory and try it again to get some more repetitions in. Practice is key!

File: scripts/shopping_cart.js (hunk 2 above)
Comment (lelandjordon) [inline scripts/shopping_cart.js:27]: `item` kept throwing an error as being undefined, but `items` didn't tigger an error.  I still can't find where `item` gets assigned any value, either.  The browser console played its Jedi mind tricks on me again.
(see final comment, too)

**Comment by lelandjordon** [comment]
After renaming the appropriate `item` and `items` tags (variables?) and still not getting it to work, I pretty much threw in the towel and compared it to the solution line-by-line.
I totally didn't catch my line 72: `totShipping = totShipping + self.calculateShipping();` was the original, buggy code, with the empty () at the end.  I've seen so many () scattered around, I'm not sure when they're supposed to contain a variable and when they're not.  The lesson and review earlier today kinda helped, but I'll need to read the lesson and my notes a few more times.

**Pull request closed without merging**